### ISSUE TITLE
`cache-read-only=true` for scheduled workflows on main branch

### ISSUE THREAD

**Issue by kkolman** [gradle/gradle-build-action#383]
The default expression for `cache-read-only` value is:

```
${{ github.ref_name != github.event.repository.default_branch }}
```

But `github.event.repository.default_branch` is not set by GA on scheduled (cron) workflow runs.
Hence `cache-read-only` is resolved to `true` not writing to cache during scheduled runs on the main branch.
For CI configrations where workflows on main line are run via scheduling exclusively (no manual workflow runs happening) cache will never be written to.

I believe this behavior is worth documenting.

Unless there's a solid replacement to `github.event.repository.default_branch` for obtaining the default branch name (didnt find one in ga contexts).

**Comment by AndrzejPw** [comment]
@kkolman you are right - that's why I changed value of cache-read-only to:
`${{ github.ref != 'refs/heads/develop' }}`

And now scheduled runs at least try to write cache. The problem is they are failing with following error:
`Save Gradle User Home failed in 'beforeSave': Error: ENOENT: no such file or directory, lstat '/Users/runner/.gradle/caches/7.4.2/kotlin-dsl/scripts/290aeeb3033fad5386fbca5f7db24fe3'`

I think it may be related to scheduled workflow runs because when I tested action triggered by PR cache was written successfully.

**Comment by bigdaz** [comment]
Thanks for the reports. I think we should be able to detect when we are running with a scheduled trigger and avoid setting `cache-read-only` in this case.

However, I'm having trouble getting a scheduled workflow to execute: I added a `ci-schedule.yml` workflow and can run it manually via `workflow_dispatch`, but it never seems to run via the cron trigger. https://github.com/gradle/gradle-build-action/actions/workflows/ci-schedule.yml

Any ideas?

**Comment by AndrzejPw** [comment]
@bigdaz I see now it is being scheduled :).  However, my error with `Save Gradle User Home failed in 'beforeSave': Error: ENOENT: no such file or directory, lstat '/Users/runner/.gradle/caches/7.4.2/kotlin-dsl/scripts/290aeeb3033fad5386fbca5f7db24fe3` is not related with scheduled runs. I tested it again and now I have this error everytime _cache-read-only_ is false.

**Comment by bigdaz** [comment]
@AndrzejPw Can you please raise a separate issue for your problem, since it appears unrelated to this reported issue? If you can provide a link to a failing build that would be great, otherwise the full stack trace would help.

**Comment by AndrzejPw** [comment]
Thank you - it does not appear anymore (maybe sth was wrong with github environment). If it happens again I will report it.

**Comment by bigdaz** [comment]
@AndrzejPw FYI there was another report of the `ENOENT: no such file or directory` from an internal project that allowed me to reproduce, diagnose and fix this issue: #408
